resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -308,7 +307,7 @@ private[spark] class ApplicationMaster(
       // The client-mode AM doesn't listen for incoming connections, so report an invalid port.
       registerAM(Utils.localHostName, -1, sparkConf,
         sparkConf.getOption("spark.driver.appUIAddress"), appAttemptId)
-      addAmIpFilter(Some(driverRef), ProxyUriUtils.getPath(appAttemptId.getApplicationId))
+      addAmIpFilter(Some(driverRef), s"/proxy/$appAttemptId")
```

Comment:
This a regression in the Hadoop APIs?

Reply:
no - this is to remove dependency on Hadoop server side (`ProxyUriUtils`, which also calls private API and is not good).